src/app/screens/connect/btcSelectAddressScreen/index.tsx

```diff
@@ -0,0 +1,292 @@
+    if (payload.network.type !== network.type) {
+      navigate('/tx-status', {
+        state: {
+          txid: '',
+          currency: 'BTC',
+          errorTitle: t('NETWORK_MISMATCH_ERROR_TITLE'),
+          error: t('NETWORK_MISMATCH_ERROR_DESCRIPTION'),
+          browserTx: true,
+        },
+      });
+    }
+  };
+
+  useEffect(() => {
+    switchAccountBasedOnRequest();
+  }, []);
+
+  useEffect(() => {
+    (async () => {
+      if (origin !== '') {
+        getAppIconFromWebManifest(origin).then((appIcons) => {
+          setAppIcon(appIcons);
+        });
+      }
+    })();
+
+    return () => {
+      setAppIcon('');
+    };
+  }, [origin]);
+
+  const AddressPurposeRow = useCallback((purpose: AddressPurpose) => {
+    if (purpose === AddressPurpose.Payment) {
+      return (
+        <AddressBox key={purpose}>
+          <AddressContainer>
+            <AddressImage src={BitcoinIcon} />
+            <AddressTextTitle>{t('BITCOIN_ADDRESS')}</AddressTextTitle>
+          </AddressContainer>
+          <TruncatedAddress>{getTruncatedAddress(btcAddress)}</TruncatedAddress>
+        </AddressBox>
+      );
+    }
+    if (purpose === AddressPurpose.Ordinals) {
+      return (
+        <AddressBox key={purpose}>
+          <AddressContainer>
+            <AddressImage src={OrdinalsIcon} />
+            <AddressTextTitle>{t('ORDINAL_ADDRESS')}</AddressTextTitle>
+          </AddressContainer>
+          <TruncatedAddress>{getTruncatedAddress(ordinalsAddress)}</TruncatedAddress>
+        </AddressBox>
+      );
+    }
+    if (purpose === AddressPurpose.Stacks) {
+      return (
+        <AddressBox key={purpose}>
+          <AddressContainer>
+            <AddressImage src={stxIcon} />
+            <AddressTextTitle>{t('STX_ADDRESS')}</AddressTextTitle>
+          </AddressContainer>
+          <div>
+            {selectedAccount?.bnsName ? <BnsName>{selectedAccount?.bnsName}</BnsName> : null}
+            <TruncatedAddress>{getTruncatedAddress(stxAddress)}</TruncatedAddress>
+          </div>
+        </AddressBox>
+      );
+    }
+
+    navigate('/tx-status', {
+      state: {
+        txid: '',
+        currency: 'BTC',
+        errorTitle: t('INVALID_PURPOSE_ERROR_TITLE'),
+        error: t('INVALID_PURPOSE_ERROR_DESCRIPTION'),
+        browserTx: true,
+      },
+    });
+  }, []);
+
+  const handleSwitchAccount = () => {
+    navigate('/account-list');
+  };
+
+  return (
+    <OuterContainer style={styles}>
+      <TitleContainer>
+        {appIcon !== '' ? <TopImage src={appIcon} alt="Dapp Logo" /> : null}
+        <Title>{t('TITLE')}</Title>
+        <DapURL>{appUrl}</DapURL>
+      </TitleContainer>
+      {payload.message ? <RequestMessage>{payload.message.substring(0, 80)}</RequestMessage> : null}
+      <SelectAccount account={selectedAccount!} handlePressAccount={handleSwitchAccount} />
+      <AddressBoxContainer>{payload.purposes.map(AddressPurposeRow)}</AddressBoxContainer>
```

Comment:
figma has a different order: payments, ordinals, stacks
and bitcoin address should be payments address
![image](https://github.com/secretkeylabs/xverse-web-extension/assets/6109710/95dc2ae6-8ef7-44cb-bad7-1681c6f859f0)

Reply:
actual: 
![image](https://github.com/secretkeylabs/xverse-web-extension/assets/6109710/69862b64-b63e-4fab-9588-51d5c0c1d716)